test/unit/global-task-list-test.js

```diff
@@ -0,0 +1,150 @@
+
+            $scope = $rootScope.$new();
+            templateHtml = $templateCache.get("/modules/tasks/global-task-detail/global-task-detail.html");
+
+            element = $compile(templateHtml)($scope);
+        });
+
+        inject(function(_utils_, _config_, _taskStore_, _clusterStore_, _globalTaskDetail_) {
+            utils = _utils_;
+            config = _config_;
+            taskStore = _taskStore_;
+            clusterStore = _clusterStore_;
+            globalTaskDetail = _globalTaskDetail_;
+        });
+
+    });
+
+    beforeEach(function() {
+        $rootScope.clusterData = globalTaskDetail.clusterList;
+        $state.current.name = "global-task-detail";
+        getClusterListDeferred = $q.defer();
+        getJobDetailDeferred = $q.defer();
+        getLogsListDeferred = $q.defer();
+        getTaskStatusDeferred = $q.defer();
+
+        sinon.stub($state, "go");
+        sinon.stub(clusterStore, "getClusterList").returns(getClusterListDeferred.promise);
+        sinon.stub(taskStore, "getJobDetail").returns(getJobDetailDeferred.promise);
+        sinon.stub(taskStore, "getTaskLogs").returns(getLogsListDeferred.promise);
+        sinon.stub(taskStore, "getTaskStatus").returns(getTaskStatusDeferred.promise);
+        clock = sinon.useFakeTimers();
+        config.msgRefreshIntervalTime = 5;
+    });
+
+    it("Should initialize all the properties", function() {
+        vm = $componentController('globalTaskDetail', { $scope: $scope });
+
+        expect($rootScope.selectedClusterOption).to.be.equal("");
+        expect(vm.clusterId).to.be.equal(globalTaskDetail.clusterList.integrationId);
+        expect(vm.isDataLoading).to.be.true;
+        expect(vm.isMessagesLoading).to.be.true;
+        expect(vm.taskDetail).to.be.an("array").that.is.empty;
+    });
+
+    describe("Task List workflows", function() {
+        beforeEach(function() {
+            vm = $componentController("globalTaskDetail", { $scope: $scope });
+            getJobDetailDeferred.resolve(globalTaskDetail.jobDetail);
+            getLogsListDeferred.resolve(globalTaskDetail.logs);
+            getTaskStatusDeferred.resolve(globalTaskDetail.status);
+            $rootScope.$digest();
+        });
+
+        it("Should fetch job details and logs", function() {
+            expect($rootScope.clusterData).to.deep.equal(globalTaskDetail.clusterList);
+            expect(vm.taskDetail).to.deep.equal(globalTaskDetail.jobDetail);
+            expect(vm.isDataLoading).to.be.false;
+            expect(vm.taskDetail.logs).to.deep.equal(globalTaskDetail.logs);
+            expect(vm.isMessagesLoading).to.be.false;
+        });
+
+        /*        it("Should go to cluster detail", function() {
+                    vm.goToClusterDetail();
+                    expect($state.go.calledWith("cluster-hosts", { clusterId: globalTaskDetail.jobDetails.parameters["TendrlContext.integration_id"] })).to.be.true;
+                });WIP
+        */
```

Comment:
@a2batic if this unit test case is not needed, Please remove it.

Reply:
Acknowledge